src/tmx/TmxUtils/src/simulation/SimulationEnvUtils.cpp

```diff
@@ -13,15 +13,17 @@ namespace tmx::utils::sim{
 
     }
 
-    std::string get_sim_config(const char *config_name) {
+    std::string get_sim_config(const char *config_name, bool required) {
         if (is_simulation_mode() && config_name) {
             try {
                 std::string config =  std::getenv(config_name);
```

Comment:
Is this the getenv() used? https://en.cppreference.com/w/cpp/utility/program/getenv#:~:text=This%20function%20is%20not%20required,obtained%20from%20a%20previous%20call. It seems to return null pointer when the config name not found?

Reply:
See unit tests. It throws a std::logic_error when configuration parameter does not exist.

Reply:
Actually I think you are right. Something about trying to assign a nullptr to a string or return it throws std::logic_error. I will fix this logic to explicitly confirm it is no null ptr.

Reply:
Addressed